### PR TITLE
[RDY] Doc cleanup (3) + terminal remnants

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: runtime/doc/vim_diff.txt [hunk 1]
```diff
@@ -114,6 +114,10 @@ Other options:
   'shelltype'
   'textauto'
   'textmode'
+  'ttybuiltin'
+  'ttymouse'
+  'ttyscroll'
+  'weirdinvert'
```
Comment (justinmk) [inline runtime/doc/vim_diff.txt]: should probably start sorting these alphanumerically

File: runtime/doc/todo.txt [hunk 2]
```diff
@@ -546,11 +546,6 @@ is confusing.  Should say "the argument list is empty".
 xterm supports escape sequences to mark a paste operation.  Need to be
 enabled. (Bruno Sutic, 2014 Jul 11)  How to know the terminal supports this?
 
-URXVT:
-- will get stuck if byte sequence does not contain the expected semicolon.
-- Use urxvt mouse support also in xterm.  Explanations:
-  http://www.midnight-commander.org/ticket/2662
```
Comment (justinmk) [inline runtime/doc/todo.txt]: @tarruda Does libtermkey or whatever report mouse coordinates using the "new way" described at http://www.midnight-commander.org/ticket/2662 ? (It's a well-written ticket worth reading, could save us time)

File: runtime/doc/various.txt [hunk 3]
```diff
@@ -277,6 +277,7 @@ g8			Print the hex values of the bytes used in the
 <			The screen is not redrawn then, thus you have to use
 			CTRL-L or ":redraw!" if the command did display
 			something.
+			Also see |:terminal|.
```
Comment (justinmk) [inline runtime/doc/various.txt]: This makes more sense on line 236. I suggest this edit:

      :!{cmd}			Execute {cmd} with 'shell'. See also |:terminal|.

File: runtime/doc/todo.txt (hunk 2 above)
Comment (Pyrohh) [inline runtime/doc/todo.txt]: As far as I'm aware, yes.

https://github.com/neovim/neovim/issues/1845#issuecomment-70988090
https://github.com/neovim/neovim/issues/1862#issuecomment-70988055

File: src/nvim/screen.c [hunk 4]
```diff
@@ -6450,7 +6450,6 @@ int screen_ins_lines (
   // - there is no valid screen
   // - the screen has to be redrawn completely
   // - the line count is less than one
```
Comment (justinmk) [inline src/nvim/screen.c]: this comment doesn't match the condition. And the entire comment is kind of redundant and/or outdated at this point.

File: src/nvim/option_defs.h [hunk 5]
```diff
@@ -567,21 +567,6 @@ EXTERN char_u   *p_titlestring; /* 'titlestring' */
 EXTERN char_u   *p_tsr;         /* 'thesaurus' */
 EXTERN int p_ttimeout;          /* 'ttimeout' */
 EXTERN long p_ttm;              /* 'ttimeoutlen' */
-EXTERN long p_ttyscroll;        /* 'ttyscroll' */
-#if defined(FEAT_MOUSE) && defined(UNIX)
-EXTERN char_u   *p_ttym;        /* 'ttymouse' */
```
Comment (justinmk) [inline src/nvim/option_defs.h]: @tarruda can you confirm that we're not going to use `'ttymouse'` and `'ttyscroll'` in the future?

File: src/nvim/option.c [hunk 6]
```diff
@@ -1581,13 +1581,12 @@ static vimoption_T
   {"ttyfast",  "tf",  P_BOOL|P_NO_MKRC|P_VI_DEF,
    (char_u *)&p_force_on, PV_NONE,
    {(char_u *)TRUE, (char_u *)0L} SCRIPTID_INIT},
+
+   // NO-OP
   {"ttymouse",    "ttym", P_STRING|P_NODEFAULT|P_NO_MKRC|P_VI_DEF,
```
Comment (justinmk) [inline src/nvim/option.c]: we should probably remove it, else we will get more tickets asking why ttymouse works in Vim but not Neovim. Already fielding multiple such issues for `t_Co` and friends.

File: runtime/doc/todo.txt (hunk 2 above)
Comment (justinmk) [inline runtime/doc/todo.txt]: :+1: That answers it

File: src/nvim/option.c (hunk 6 above)
Comment (Pyrohh) [inline src/nvim/option.c]: :+1: I hadn't considered that.

File: runtime/doc/various.txt (hunk 3 above)
Comment (Pyrohh) [inline runtime/doc/various.txt]: :+1:

File: src/nvim/screen.c (hunk 4 above)
Comment (Pyrohh) [inline src/nvim/screen.c]: No clue if it's outdated, but it does seem redundant. I removed it.

**Comment by Pyrohh** [comment]
This has been stagnating for a while because I haven't bothered dealing with the last commit (which was regarding 2-3 options). I've narrowed that down to just removing `-T` and marked as RFC.

**Comment by Pyrohh** [comment]
Fixed the merge conflict and rebased. Could you take a look at this @glts?

**Comment by glts** [comment]
I will after a night's rest :)

**Comment by glts** [comment]
Just took a quick look. I can't really comment on this pull request as I have no idea about Neovim's tty stuff or mouse handling and I'm not in the loop about these 'terminal remnants', sorry.

**Comment by Pyrohh** [comment]
> I can't really comment on this pull request as I have no idea about Neovim's tty stuff or mouse handling and I'm not in the loop about these 'terminal remnants', sorry.

It's fine, I appreciate you taking a look anyways.

@justinmk Sorry to keep doing this, but could you take a look? Not many people seem to be aware of the legacy terminal infrastructure.

**Comment by Pyrohh** [comment]
Added one more small commit as discussed in https://github.com/neovim/neovim/pull/2609#discussion_r29975613.

File: src/nvim/option_defs.h (hunk 5 above)
Comment (Pyrohh) [inline src/nvim/option_defs.h]: @tarruda now that you're back, could you please answer this :-)

**Comment by Pyrohh** [comment]
Fixed the merge conflict and rebased.

**Comment by Pyrohh** [comment]
I added more info to all the commit messages and repushed. Hopefully that makes reviewing this easier.

**Comment by Pyrohh** [comment]
Rebasing changes to `runtime/syntax/vim.vim` is a nightmare due to the (very) long lines, so I split those changes out in to one commit.

@justinmk @fwalch could you take a look please?

**Comment by justinmk** [comment]
Looking now

File: runtime/doc/options.txt [hunk 7]
```diff
@@ -6771,80 +6771,16 @@ A jump table for the options with a short description can be found at |Q_op|.
...
+'ttybuiltin' 'tbi'	Removed. |vim-differences| {Nvim}
 
 				     *'ttyfast'* *'tf'* *'nottyfast'* *'notf'*
 'ttyfast' 'tf'		Removed. |vim-differences| {Nvim}
 
 						*'ttymouse'* *'ttym'*
-'ttymouse' 'ttym'	string	(default depends on 'term')
-			global
-			{only in Unix, doesn't work in the GUI; not
-			available when compiled without |+mouse|}
```
Comment (justinmk) [inline runtime/doc/options.txt]: This ("only on unix, not in the GUI") is a good hint that it's not an option we want to keep. 

Platform-dependent features should be implemented as providers and/or passive events, possibly with lowest-common denominator support existing in the core (as is the case with TUI mouse handling).

File: runtime/doc/starting.txt [hunk 8]
```diff
@@ -772,7 +765,7 @@ To change it later, use a ":let" command like this: >
 In case you need the value of $VIMRUNTIME in a shell (e.g., for a script that
 greps in the help files) you might be able to use this: >
 
-	VIMRUNTIME=`vim -e -T dumb --cmd 'exe "set t_cm=\<C-M>"|echo $VIMRUNTIME|quit' | tr -d '\015' `
+	VIMRUNTIME=`vim -e --cmd 'exe "set t_cm=\<C-M>"|echo $VIMRUNTIME|quit' | tr -d '\015' `
```
Comment (justinmk) [inline runtime/doc/starting.txt]: This can be:

    nvim -e --cmd 'echo $VIMRUNTIME|quit' | tr -d '\015'

**Comment by justinmk** [comment]
LGTM, try squashing to 1 or 2 commits though.

File: runtime/doc/options.txt (hunk 7 above)
Comment (Pyrohh) [inline runtime/doc/options.txt]: :+1:

File: runtime/doc/starting.txt (hunk 8 above)
Comment (Pyrohh) [inline runtime/doc/starting.txt]: :+1: Although that didn't work for me because it was sending it to standard error. See the new commit.

**Comment by Pyrohh** [comment]
Mkay, got it from 8 to 5 commits, with the only difference in the commit messages and that example you commented on.

I'll merge after the tests.

**Pull request merged**